Add option to explicitly include or exclude specific nodes

ttarczynski opened the pull request:
## Pull Request Checklist

**Is this in reference to an existing issue?**

#### General

- [x] Update Changelog following the conventions laid out at [here ](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md)

- [x] Update README with any necessary configuration snippets

- [ ] Binstubs are created if needed

- [x] RuboCop passes

- [x] Existing tests pass

#### New Plugins

N/A

#### Purpose

Adds the ability to explicitly include or exclude nodes in check-kube-pods-pending/restarting/running, check-kube-nodes-ready. It is useful when some nodes are intentionally NotReady but we still need to check what's running on other nodes.
I've followed the pattern used in:
- #37 "ability to explicitly include namespaces"

#### Known Compatibility Issues

majormoses commented:
Thanks for your contribution to Sensu plugins! Without people like you submitting PRs we couldn't run the project. I will review it shortly.

majormoses commented on CHANGELOG.md:
```diff
@@ -4,6 +4,8 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
+### Changed
+ - Add option to explicitly include or exclude specific nodes (@ttarczynski)
```
I think it would be nice if we include the option names, the scripts that were updated, and a bit of usage information. For example:
```
- `check-kube-nodes-ready.rb`, `check-kube-pods-pending.rb`, `check-kube-pods-restarting.rb`, `check-kube-pods-running.rb`: added options `--included-nodes` and `--excluded-nodes` which support a comma separated list so you can filter nodes (@yourgithubusername)
```

ttarczynski commented:
@majormoses thank you for the review!
* I've applied changes in options naming and docs
* will get back to this next week and try to follow your advise on lib and the exclude logic

majormoses commented:
Apologies in advance if next week I do not respond quickly as I will be traveling for work

ttarczynski commented on bin/check-kube-nodes-ready.rb:
```diff
@@ -56,4 +72,9 @@ def run
   end
+
+  def should_exclude_node(node_name)
+    return !config[:include_node].include?(node_name) unless config[:include_node].empty?
```
I've change the logic a little bit and added info on how it works to docs:
`Exclude wins when a node is in both include and exclude lists`

ttarczynski commented:
@majormoses i've just applied the remaining fixes:
- moved the 'should_exclude_node' function to a lib
- changed the logic when a node is in both include and exclude lists and added info in docs

majormoses commented:
Thanks, I will review this evening.

majormoses commented:
released: https://rubygems.org/gems/sensu-plugins-kubernetes/versions/3.1.0
